common/djangoapps/student/models.py

```diff
@@ -504,8 +504,29 @@ def register(self, user):
 
     def activate(self):
         self.user.is_active = True
+        self._track_activation()
         self.user.save()
 
+    def _track_activation(self):
+        """ Update the isActive flag in mailchimp for activated users."""
+        has_segment_key = hasattr(settings, 'LMS_SEGMENT_KEY') and settings.LMS_SEGMENT_KEY
+        has_mailchimp_id = hasattr(settings, 'MAILCHIMP_NEW_USER_LIST_ID')
+        if has_segment_key and has_mailchimp_id:
+            identity_args = [
+                self.user.id,  # pylint: disable=no-member
+                {
+                    'email': self.user.email,
+                    'username': self.user.username,
+                    'isActive': 1,
```

Comment:
I don't see this "trait" described in Segment's MailChimp https://segment.com/docs/integrations/mailchimp/ docs. Is this a custom MailChimp attribute we've created? If so, I'd add a comment explaining what's going on.